### DISCUSSION TITLE
Stenographer Only Recording Broadcast/Multicast Traffic

### DISCUSSION THREAD

**Discussion by dayton3375** [Security-Onion-Solutions/securityonion#5853] [Unsupported Versions] [answered]
I've been absolutely stumped as to what could be causing this. Stenographer is only recording traffic that is destined to broadcast or multi-cast traffic (ARP, Cisco Discovery Protocols), however when I run Wireshark on the Bond0 interface that it's configured to record from, it clearly shows all the traffic passing through (not just broadcast/multicast). Also I have Zeek running and it records all traffic too, so for some reason Stenographer seems to be the only thing that can't see all the traffic or is somehow only recording broadcast/multicast. 

I've installed the latest Standalone Security Onion 2.3.80 at this time, and I've tried re-installing it many times. Just to get more of an understanding of how I'm using Security Onion is I am doing a study where I have it monitoring a network tap in front of my router just to see how much malicious traffic hits my router every day. Security Onion is running in a VirtualBox with promiscuous mode on for the bridged monitor interface. Again, with Wireshark monitoring the Bond0 interface in Security Onion, I am able to see all the traffic, Zeek is recording all the traffic metadata, however Stenographer is only recording broadcast/multicast traffic. This is the default installation, and I've tried many times re-installing. Might there be something I'm missing here? I really appreciate any advice. I'd imagine maybe someone else has encountered this issue.

Here's some configuration information below if it is helpful:

Stenographer configuration at /opt/so/conf/steno/config
{
  "Threads": [
    { "PacketsDirectory": "/nsm/pcap"
    , "IndexDirectory": "/nsm/pcapindex"
    , "MaxDirectoryFiles": 30000
    , "DiskFreePercentage": 10
    }
  ]
  , "StenotypePath": "/usr/bin/stenotype"
  , "Interface": "bond0"
  , "Port": 1234
  , "Host": "127.0.0.1"
  , "Flags": ["-v", "--uid=stenographer", "--gid=stenographer"]
  , "CertPath": "/etc/stenographer/certs"
}

Manager configuration: /opt/so/saltstack/local/pillar/global.sls
global:
  soversion: '2.3.80'
  hnmanager: '64.126.128.0/18'
  dockernet: '172.17.0.0'
  mdengine: 'ZEEK'
  ids: 'Suricata'
  url_base: '192.168.88.189'
  managerip: '192.168.88.189'
  airgap: False
  hivekey: 'Kj3fwp6o3uRDxtDj1ddi'
  hiveplaysecret: 'zCYxuG3fxoFNtgIB3EMV'
  cortexkey: '7OBsvPgoWMuMZDLjYcdA'
  cortexorgname: 'SecurityOnion'
  cortexorguserkey: 'LGXxRUF7Ztz7qCAP8fWu'
  cortexplaysecret: 'I4QqGQqo861GlljOM5Lt'
  fleet_custom_hostname:
  fleet_manager: True
  fleet_node: False
  fleet_packages-timestamp: '2021-10-06-23:22'
  fleet_packages-version: 2
  fleet_hostname: 'myonionsensor'
  fleet_ip: '192.168.88.189'
  sensoronikey: 'U1dAGP5w470Ad2B7UuMl'
  wazuh: 1
  imagerepo: 'security-onion-solutions'
  pipeline: 'redis'
sensoroni:
  node_checkin_interval_ms: 1000
strelka:
  enabled: 1
  rules: 1
  repos:
    - 'https://github.com/Neo23x0/signature-base'
curator:
  hot_warm: False
elastic:
  features: False
elasticsearch:
  true_cluster: False
  replicas: 0
  discovery_nodes: 1
  hot_warm_enabled: False
  cluster_routing_allocation_disk.threshold_enabled: true
  cluster_routing_allocation_disk_watermark_low: '95%'
  cluster_routing_allocation_disk_watermark_high: '98%'
  cluster_routing_allocation_disk_watermark_flood_stage: '98%'
  index_settings:
    so-beats:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-firewall:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-flow:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-ids:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-import:
      shards: 1
      warm: 7
      close: 73000
      delete: 73001
    so-osquery:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-ossec:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-strelka:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-syslog:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-zeek:
      shards: 5
      warm: 7
      close: 45
      delete: 365
minio:
  access_key: ''
  access_secret: ''
s3_settings:
  size_file: 2048
  time_file: 1
  upload_queue_size: 4
  encoding: 'gzip'
  interval: 5
backup:
  locations:
    - /opt/so/saltstack/local
soctopus:
  playbook:
    rulesets:
      - windows
docker:
  range: '172.17.0.0/24'
  bip: '172.17.0.1/24'
redis_settings:
  redis_maxmemory: 812
playbook:
  api_key: f79ac7aedbad418effe44942b816880b35bc7b62

My Sensor Configuration at /opt/so/saltstack/local/pillar/minions/myonionsensor_standalone.sls
host:
  mainint: 'enp0s3'
sensoroni:
  node_address: '192.168.88.189'
  node_description: ''

ntp:
  servers:
    - '0.pool.ntp.org'
    - '1.pool.ntp.org'
sensor:
  interface: 'bond0'
  mainip: '192.168.88.189'
  mainint: 'enp0s3'
  zeek_lbprocs: 2
  suriprocs: 2
  manager: ''
  mtu: 1500
  uniqueid: 1633561111
  hnsensor:
steno:
  enabled: True
patch:
  os:
    source: 'direct'
    schedule_name: 'auto'
    enabled: True
    splay: 300

manager:
  mainip: '192.168.88.189'
  mainint: 'enp0s3'
  proxy: ''
  no_proxy: ''
  esheap: '5421m'
  esclustername: '{{ grains.host }}'
  freq: 0
  domainstats: 0
  mtu: 1500
  elastalert: 1
  es_port: 9200
  cur_close_days: 30
  grafana: 1
  osquery: 1
  thehive: 1
  playbook: 1

elasticsearch:
  mainip: '192.168.88.189'
  mainint: 'enp0s3'
  esheap: '5421m'
  esclustername: '{{ grains.host }}'
  node_type: ''
  es_port: 9200
  log_size_limit: 164
  node_route_type: 'hot'

logstash_settings:
  ls_pipeline_batch_size: 125
  ls_input_threads: 1
  lsheap: 1000m
  ls_pipeline_workers: 6

idstools:
  config:
    ruleset: 'ETOPEN'
    oinkcode: ''
    urls:
  sids:
    enabled:
    disabled:
    modify:

kratos:
  kratoskey: 'm1sL2b837Pl1q5VdTwgK'

sensoroni.json:
{
  "logFilename": "/opt/sensoroni/logs/sensoroni.log",
  "logLevel":"info",
  "agent": {
    "nodeId": "myonionsensor",
    "role": "so-standalone",
    "description": "",
    "address": "192.168.88.189",
    "model": "",
    "pollIntervalMs": 1000,
    "serverUrl": "https://192.168.88.189/sensoroniagents",
    "verifyCert": false,
    "modules": {
      "importer": {},
      "statickeyauth": {
        "apiKey": "U1dAGP5w470Ad2B7UuMl"
      },
      "stenoquery": {
        "executablePath": "/opt/sensoroni/scripts/stenoquery.sh",
        "pcapInputPath": "/nsm/pcap",
        "pcapOutputPath": "/nsm/pcapout"
      }
    }
  }
}

<img width="468" alt="Packets" src="https://user-images.githubusercontent.com/45369576/136969567-12c7eb01-e45c-467d-b0ab-e94de6147c25.png">

**Comment by dayton3375** [comment, marked as answer]
I sort of fixed this I believe by changing the interface from Bond0 to the monitor interface and restarting the salt configuration and stenographer like in the documentation. However, I discovered there's a bug where I can't pivot from Kibana to PCAP by clicking the ID like it is suppose too. The only way for me to view the packets of an event is to go to PCAP in the SOC and query the time frame that it occurred, then sifting through the packets to locate the malicious packet.

**Reply by dougburks** [reply]
This is really strange. I don't believe we've had any reports like this before. I would lean towards it being some strange interaction between the NIC on your host, Virtualbox, and the VM. I suspect if you installed on bare hardware, this would not happen.

**Reply by dayton3375** [reply]
That actually makes sense. I'll try it on an actual computer when I get one and see if it works like it's suppose to. Thank you for the feedback.

**Reply by dayton3375** [reply]
If it's useful to know, I tried installing Security Onion in Hyper-V instead of VirtualBox and got the same results. In Kibana or Alerts I can only pivot to the PCAP of the log if it is a packet to a multicast or broadcast address such as arp. Any other traffic I have to add a job for the time and then look in that packet capture for the packet of interest. Still odd how on both virtualbox and hyper-v it will only view the PCAP for multicast traffic when clicking on the ID, but the rest it shows "No data to show" unless requesting an actual job for the timeframe. Maybe this pivoting feature will only work on bare metal and no virtual machines.

**Reply by dougburks** [reply]
It's not an issue with virtual machines as a whole as we have many virtualized test and production deployments that work just fine.

Since you get the same results with Hyper-V, that would seem to eliminate VirtualBox as a potential culprit. My next step would be installing Security Onion on bare hardware. Also check your NIC and replace with an Intel NIC if necessary.

You said above that you're monitoring a network tap. Is this a standard tap or could it be encoding the traffic in some way?

**Reply by cbisaccia78** [reply]
Can confirm that this issue also happens on bare hardware.

**Reply by dayton3375** [reply]
Actually the issue is mostly gone for me now oddly. I think the new release 2.3.100 somehow fixed it, however it still doesn't pivot to PCAP for some traffic like when it's just a ping. So still the only way to get the pcap for that kind of traffic is to just use to timestamp to request the pcap for everything around that time. All other traffic sessions pivot to pcap perfectly now for me though and I'm still using hyper-v. The network tap I use is called SharkTap and it doesn't do any encoding or anything. Just purely copies all packets that pass through it.

**Reply by dougburks** [reply]
@cbisaccia78 If you would like to troubleshoot your issue further, please start a new discussion rather than replying to someone else's discussion.
